tests/src/test/scala/quasiquotes/QuasiquoteSuite.scala

```diff
@@ -306,4 +392,286 @@ class QuasiquoteSuite extends FunSuite {
     val q"$x" = q"42"
     assert(x.show[Syntax] === "42")
   }
+
+  test("arg\"$name = $expr\"") {
+    val name = q"x"
+    val expr = q"foo"
+    assert(arg"$name = $expr".show[Syntax] === "x = foo")
+  }
+
+  test("arg\"$expr: _*\"") {
+    val expr = q"foo"
+    assert(arg"$expr: _*".show[Syntax] === "foo: _*")
+  }
+
+  test("arg\"$expr\"") {
+    val expr = q"foo"
+    assert(q"$expr".show[Syntax] === "foo")
+  }
+
+  test("t\"$ref.$tname\"") {
+    val ref = q"X"
+    val tname = t"Y"
+    assert(t"$ref.$tname".show[Structure] === "Type.Select(Term.Name(\"X\"), Type.Name(\"Y\"))")
+  }
+
+  test("t\"$tpe#$tname\"") {
+    val tpe = t"X"
+    val tname = t"Y"
+    assert(t"$tpe#$tname".show[Syntax] === "X#Y")
+  }
+
+  test("t\"$ref.type\"") {
+    val ref = q"X"
+    assert(t"$ref.type".show[Syntax] === "X.type")
+  }
+
+  test("t\"$tpe[..$tpes]") {
+    val tpe = t"X"
+    val tpes = List(t"Y", t"Z")
+    assert(t"$tpe[..$tpes]".show[Syntax] === "X[Y, Z]")
+  }
+
+  test("t\"$tpe $tname $tpe\"") {
+    val tpe1 = t"X"
+    val tname = t"Y"
+    val tpe2 = t"Z"
+    assert(t"$tpe1 $tname $tpe2".show[Syntax] === "X Y Z")
+  }
+
+  test("t\"(..$atpes) => $tpe\"") {
+    val atpes = List(t"X", t"Y")
+    val tpe = t"Z"
+    assert(t"(..$atpes) => $tpe".show[Syntax] === "(X, Y) => Z")
+  }
+
+  test("t\"(..$tpes)\"") {
+    val tpes = List(t"X", t"Y")
+    assert(t"(..$tpes)".show[Syntax] === "(X, Y)")
+  }
+
+  test("1 t\"..$tpes { ..$stats }\"") {
+    val t"..$tpes {..$stats}" = t"A with B with C { val a: A; val b: B }"
+    assert(tpes.toString === "List(A, B, C)")
+    assert(tpes(0).show[Syntax] === "A")
+    assert(tpes(1).show[Syntax] === "B")
+    assert(tpes(2).show[Syntax] === "C")
+    assert(stats.toString === "List(val a: A, val b: B)")
+    assert(stats(0).show[Syntax] === "val a: A")
+    assert(stats(1).show[Syntax] === "val b: B")
+  }
+
+  test("2 t\"..$tpes { ..$stats }\"") {
+    val tpes = List(t"X", t"Y")
+    val stats = List(q"val a: A", q"val b: B")
+    assert(t"..$tpes { ..$stats }".show[Syntax] === "X with Y { val a: A; val b: B }")
+  }
+
+  test("t\"$tpe forSome { ..$stats }\"") {
+    val tpe = t"X"
+    val stats = List(q"val a:A", q"val b:B")
+    assert(t"$tpe forSome { ..$stats }".show[Syntax] === "X forSome { val a: A; val b: B }")
+  }
+
+  test("t\"$tpe ..@$annots\"") {
+    val tpe = t"X"
+    val annots = List(mod"@a", mod"@b")
+    assert(t"$tpe ..@$annots".show[Syntax] === "X @a @b")
+  }
+
+  test("t\"_ >: $tpeopt <: $tpeopt\"") {
```

Comment:
Probably I should add one more comment
`// TODO change to expropt (and test it) after issue #199 resolved`